__tests__/String/trim.test.ts

```diff
@@ -0,0 +1,13 @@
+import { expectType } from 'ts-expect'
+
+import { S } from '../..'
+
+describe('trim', () => {
+  it('provides correct types', () => {
+    expectType<string>(S.trim('text'))
+  })
+
+  it('returns a new string with leading and trailing whitespace removed from string', () => {
+    expect(S.trim('  text')).toEqual('text')
+  })
+})
```

Comment:
could you please add a test with the `*` string as a first argument of `it`? 🙏 take a look at the following test: https://github.com/mobily/ts-belt/blob/master/__tests__/Bool/and.test.ts#L16? (it automatically adds an example in the documentation)